web-server/pages/api/internal/version.ts

```diff
@@ -103,25 +104,61 @@ async function fetchDockerHubTags(): Promise<
   }));
 }
 
+async function fetchLatestGitHubCommit(): Promise<GitHubCommit> {
+  const apiUrl = `https://api.github.com/repos/${githubOrgName}/${githubRepoName}/commits`;
+  const response = await axios.get<GitHubCommit[]>(apiUrl);
+  const latestCommit = response.data[0];
+  return latestCommit;
+}
+
+function isUpdateAvailable(
+  localVersionInfo: ProjectVersionInfo,
+  dockerLatestRemoteTag: TagCompressed,
+  githubLatestCommit: GitHubCommit
```

Comment:
[optional]
when passing more than two arguments, its wise to wrap the args into an object, gives you context and control over the args.
```
const isUpdateAvailable = (
  {
     localVersionInfo,
     dockerLatestRemoteTag,
     githubLatestCommit
  }:
  {
     localVersionInfo: ProjectVersionInfo;
     dockerLatestRemoteTag: TagCompressed;
     githubLatestCommit: GitHubCommit;
  }) => {
  ...
  }
  ```